include/deal.II/grid/grid_generator.h

```diff
@@ -2367,6 +2367,24 @@ namespace GridGenerator
    * the surface mesh. If left at its default, i.e., if the set is empty, then
    * the function operates on <i>all</i> boundary faces.
    *
+   * To prevent printing a very long type in the doxygen documentation the
+   * actual return type of this function is
+   *
+   * @code
+   * std::map<typename DoFHandler<dim - 1, spacedim>::cell_iterator,
+   *          typename DoFHandler<dim, spacedim>::face_iterator>
+   * @endcode
+   *
+   * when <tt>MeshType</tt> is DoFHandler and
+   *
+   * @code
+   * std::map<typename Triangulation<dim - 1, spacedim>::cell_iterator,
+   *          typename Triangulation<dim, spacedim>::face_iterator>
+   * @endcode
```

Comment:
doxygen doesn't like using MeshType here since it cannot link it to anything else so I had to do a workaround.